### PR TITLE
Bank statement report

### PR DESCRIPTION
**Pull request by irwalker**
Adds client support for hidden (probably deprecated) Xero bank_statement report. Needed for our cash account sync with Xero. This is a temporary solution, Ben is going to handle seeing if we can get a better solution working with Xero. We may have to deprecate our cash account sync if this functionality is going to go away.

Sorry for the spec diff, a lot of lines were adjusted by my editor as the library we've forked is auto-generated code 😬 

[vimaly](https://vimaly.com/#j8mqm/t/xero_oauth2_2._Use_new_OAuth2_authentication_schem.../wDDbNQ8D6lcnhj77E)

### CODE REVIEW

File: spec/accounting/api/accounting_api_spec.rb [hunk 1]
```diff
@@ -32,6 +32,53 @@
     end
   end
 
+  describe 'bank statement report' do
```
Comment (alto) [inline spec/accounting/api/accounting_api_spec.rb:35]: Must feel good to write some rspec again 😬

**Review by alto — APPROVED** [review]
LGTM

Do we plan to create a pull request against the original gem for this? If yes, we need to change some things to make it acceptable (remove `vcr`, do not change the specs aside from the actual test addition).

**Comment by irwalker** [comment]
No plan to merge this back in @alto . From what I can tell (community forum posts) this endpoint has been quietly deprecated and removed from documentation. So we want the fact we are using it to be slightly under the radar 😝 .

Ultimately we will have to either work with Xero to come up with a new way of getting this data, or deprecate the xero-sync cash account functionality, at which point we can archive this fork when the functionality is removed.

File: spec/accounting/api/accounting_api_spec.rb (hunk 1 above)
Comment (CitationNerded) [reply spec/accounting/api/accounting_api_spec.rb:35]: look at these tests. bloody art. :tada:

**Review by CitationNerded — APPROVED** [review]
LGTM - good commit breakdown

**Pull request merged**